Source/WebCore/page/NavigationHistoryEntry.h

```diff
@@ -42,26 +43,26 @@ class NavigationHistoryEntry final : public RefCounted<NavigationHistoryEntry>,
     using RefCounted<NavigationHistoryEntry>::ref;
     using RefCounted<NavigationHistoryEntry>::deref;
 
-    static Ref<NavigationHistoryEntry> create(ScriptExecutionContext* context) { return adoptRef(*new NavigationHistoryEntry(context)); }
+    static Ref<NavigationHistoryEntry> create(ScriptExecutionContext* context, const URL& url) { return adoptRef(*new NavigationHistoryEntry(context, url)); }
 
-    const String& url() const { return m_url; };
-    const String& key() const { return m_key; };
-    const String& id() const { return m_id; };
+    const String& url() const { return m_url.string(); };
+    String key() const { return m_key.toString(); };
```

Comment:
In my later patch this is called in loops to lookup entries. Maybe the string could be cached, or the primary way it's stored?

Reply:
Yeah, this is a first stab at the key/id handling, it can definitely change later, as you said for example for efficiency. I mainly added it for now for identification/debug reasons (so far that means through WebKit inspector, probably later for WPT tests).